complete/src/main/java/price/checker/service/StockPriceRpsyService.java

```diff
@@ -0,0 +1,121 @@
+package price.checker.service;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import price.checker.domain.rpsy.StockPriceRepository;
+import price.checker.domain.StockPrice;
+import price.checker.entrypoint.exception.EntityNotFoundException;
+
+import javax.swing.text.html.parser.Entity;
+import java.io.IOException;
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.*;
+import java.util.stream.Collector;
+import java.util.stream.Collectors;
+
+/**
+ * This is the service responsible for updating, deleting, retrieving
+ * and adding new data entries into mySQL table "StockPrice"
+ */
+
+@Slf4j
+@Service
+public class StockPriceRpsyService {
+
+    @Autowired
+    private StockPriceRepository stockPriceRepository;
+
+    private final DateFormat DF = new SimpleDateFormat("yyyy-MM-dd");
+    /**
+     * This is the method that is used to check inside our database
+     * to see if we have all the data wanted by the user available
+     * 1 = yes
+     * 0 = no
+     * @param symbol
+     */
+    public Integer checkForWantedPrices(String symbol)throws ParseException, IOException {
+        return stockPriceRepository.findMostRecent(symbol);

```

Comment:
You should have this return a boolean instead, change this line to:

        return stockPriceRepository.findMostRecent(symbol) == 1 ? true : false;

If this method returns a boolean, you should also change the method name to a question, something like "doesPriceExist()"

Reply:
or a statemeant that could be proven false like --> "priceExists()"